src/test/java/org/springframework/data/hazelcast/repository/query/QueryIT.java

```diff
@@ -86,10 +86,45 @@
     private PersonRepository personRepository;
 
     // Count methods
+    @Override
+    public void setUp() {
+        System.out.println("setUp()");
```

Comment:
Please don't add any extra logs like that

Reply:
I created this PR to show my progress, no need to merge yet, I modified just one method countByFirstname() to show my progress, Could you tell me is it a right way to solve the issue?

Reply:
There's no easy answer to this question. So far so good, but it might get more complicated along the way. 

But yes, test cases should be independent of the underlying state and work just fine with empty maps.

Reply:
thank you. I'll continue then

Reply:
A tip for the future - GitHub allows you to submit drafts of PullRequests